### ISSUE TITLE
Data not uploaded to storage and index not created due to errors with AzureDeveloperCliCredential not found and SSL certificate error

### ISSUE THREAD

**Issue by SonalPrabhune** [Azure-Samples/azure-search-openai-demo#97]
(Auto-closed as obsolete; see final comment for details).

**Comment by vhvb1989** [comment]
> had to be installed explicitly

There a `.sh` and `.ps1` file within the /scripts folder to launch the `prepdocs.py`
These scripts take care of creating a python virtual environment and installing all dependencies before running the python app.

If you are using Windows, try running the `ps1` file from `powershell7`

**Comment by SonalPrabhune** [comment]
Yes that’s the one I have been using all along the power shell script .ps1. Why do I still keep getting the certificate error?

**Comment by vhvb1989** [comment]
The error on your logs:

```
subprocess.CalledProcessError: Command '['cmd', '/c', 'az account get-access-token --output json --resource https://search.azure.com --tenant ']' returned non-zero exit status 1.
```

It looks like there's a call to `az` which is failing.  Do you have `az` installed?  

Note: `az` and `azd` does not share the same authentication.  This means that if you have `az` login to another account, it might be failing when it is been called.  You can try running `az logout` so it is not used as source of auth.  Then try running again.

**Comment by pauldj54** [comment]
Hi @SonalPrabhune , 
I'm running into the same issues. I would like to ask you if you managed to solve them?

**Comment by MSavka** [comment]
Hi guys, if someone still has the same issue or #79, and you have the "Owner" role assigned, etc. - the problem is in the ".../scripts/prepdocs.ps1" file, line 39, actually here:

`$cwd = (Get-Location)`
`Start-Process -FilePath $venvPythonPath -ArgumentList "./scripts/prepdocs.py $cwd/data/* --storageaccount ...`

, and this parameter: `$cwd/data/*` in the code above. You can quickly notice (Windows OS) that if your folder/path with the project contains spaces, the whole line will be parsed incorrectly and not executed (but this Python script actually uploads PDF documents in chunks and creates the necessary search index).

So the easiest fix - is to move the project into some folder without spaces (whole path), wrap this parameter with some quotes, or use a variable. Also (ideally), you need to replace slashes in the opposite direction for this path (so it will be `$cwd\data\*` ). That's all, hope it will help somebody )

**Comment by pamelafox** [comment]
Closing as obsolete.

Thanks for reporting this originally and to everyone who added context. The ingestion and indexing flow has since been updated:
- Current `scripts/prepdocs.ps1` / `scripts/prepdocs.sh` create the virtual environment, install all dependencies, and invoke `prepdocs.py` directly (no manual per‑package installs needed).
- The PowerShell script now quotes the data glob ("$cwd/data/*"), avoiding the earlier Windows path + spaces parsing issue noted in later comments.
- `AzureDeveloperCliCredential` is now included and referenced consistently across scripts and tests; the original ImportError should no longer occur with the present dependency set.
- The remaining SSL certificate failures were environmental (corporate proxy / self‑signed cert trust chain) rather than a defect in the sample. Guidance for proxy/cert handling is covered in Azure CLI docs linked in the original error output.

If you can still reproduce a failure with the latest main branch and a trusted certificate chain, please open a new issue with:
1. Exact command you ran (script vs direct python invocation)
2. OS + shell
3. `az version` output and whether `az login` succeeds independently
4. Any proxy or custom cert details (sanitized)

Appreciate the contribution—feel free to open a fresh issue if something similar crops up in the current codebase.
